### PR TITLE
@ #34 | support gatling-rsync auto recovery

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #34 
please check this PR! Thank you!

### CODE REVIEW

**Comment by hoatle** [comment]
let's check this on teracy-dev v0.6.0-a5 with the following config section on the teracy-dev-entry/config_default.yaml file with different following scenarios:


1 - vagrant-gatling-rsync plugin is not installed => expect: do nothing

2 - vagrant-gatling-rsync plugin installed but no vagrant-gatling-rsync config => expect: do nothing

3 - vagrnat-gatling-rsync plugin is installed with the following config_default.yaml file:

```yaml
vagrant:
  plugins:
    - _id: "entry-gatling-rsync"
      name: vagrant-gatling-rsync
      version: "" # empty means latest by default
      state: installed # uninstalled
      enabled: true
      config_key: gatling
      options:
        latency: 0.5
        time_format: "%%H:%%M:%%S"
        rsync_on_startup: true


teracy-dev:

  extensions:

    - _id: "entry-essential"
      path:
        extension: teracy-dev-essential
      location:
        git:
          remote:
            origin: https://github.com/teracyhq-incubator/teracy-dev-essential.git
          branch: develop
      require_version: ">= 0.3.0-SNAPSHOT"
      enabled: true


nodes:
  - _id: "0" # core-0
    vm:
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
  - _id: "entry-1"
    name: "%{node_name_prefix}-02"
```

=> expect: gatling-rsyn-auto should run on the node with _id: "0"

4. 

```yaml
vagrant:
  plugins:
    - _id: "entry-gatling-rsync"
      name: vagrant-gatling-rsync
      version: "" # empty means latest by default
      state: installed # uninstalled
      enabled: true
      config_key: gatling
      options:
        latency: 0.5
        time_format: "%%H:%%M:%%S"
        rsync_on_startup: false


teracy-dev:

  extensions:

    - _id: "entry-essential"
      path:
        extension: teracy-dev-essential
      location:
        git:
          remote:
            origin: https://github.com/teracyhq-incubator/teracy-dev-essential.git
          branch: develop
      require_version: ">= 0.3.0-SNAPSHOT"
      enabled: true


nodes:
  - _id: "0" # core-0
    vm:
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
  - _id: "entry-1"
    name: "%{node_name_prefix}-02"
```

in this case, `rsync_on_startup: false` => expect: do nothing

5. 

```yaml
vagrant:
  plugins:
    - _id: "entry-gatling-rsync"
      name: vagrant-gatling-rsync
      version: "" # empty means latest by default
      state: installed # uninstalled
      enabled: true
      config_key: gatling
      options:
        latency: 0.5
        time_format: "%%H:%%M:%%S"
        rsync_on_startup: true


teracy-dev:

  extensions:

    - _id: "entry-essential"
      path:
        extension: teracy-dev-essential
      location:
        git:
          remote:
            origin: https://github.com/teracyhq-incubator/teracy-dev-essential.git
          branch: develop
      require_version: ">= 0.3.0-SNAPSHOT"
      enabled: true


nodes:
  # - _id: "0" # core-0
  #   vm:
  #     synced_folders:
  #       - _id: "entry-workspace"
  #         type: rsync
  #         host: "./workspace"
  #         guest: "/home/vagrant/workspace"
  #         rsync__exclude:
  #           - ".git"
  #           - ".idea"
  #           - ".#*"
  #         create: true
  - _id: "entry-1"
    name: "%{node_name_prefix}-02"
    vm:
      hostname: "%{node_hostname_prefix}-02.%{node_domain_affix}"
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
```

=> expect: gatling-rsyn-auto should run on the node with _id: "entry-1"


6.

```yaml
vagrant:
  plugins:
    - _id: "entry-gatling-rsync"
      name: vagrant-gatling-rsync
      version: "" # empty means latest by default
      state: installed # uninstalled
      enabled: true
      config_key: gatling
      options:
        latency: 0.5
        time_format: "%%H:%%M:%%S"
        rsync_on_startup: true


teracy-dev:

  extensions:

    - _id: "entry-essential"
      path:
        extension: teracy-dev-essential
      location:
        git:
          remote:
            origin: https://github.com/teracyhq-incubator/teracy-dev-essential.git
          branch: develop
      require_version: ">= 0.3.0-SNAPSHOT"
      enabled: true


nodes:
  - _id: "0" # core-0
    vm:
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
  - _id: "entry-1"
    name: "%{node_name_prefix}-02"
    vm:
      hostname: "%{node_hostname_prefix}-02.%{node_domain_affix}"
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
```

=> expect: gatling-rsync-auto should run on both nodes

@hieptranquoc make sure the changes should pass the above scenarios.

**Comment by hoatle** [comment]
assigned to @datphan to review

**Comment by hoatle** [comment]
case 7: by default, `rsync_on_startup` is `true` (follows the vagrant-gatling-rsync's behavior)

```yaml
vagrant:
  plugins:
    - _id: "entry-gatling-rsync"
      name: vagrant-gatling-rsync
      version: "" # empty means latest by default
      state: installed # uninstalled
      enabled: true
      config_key: gatling
      options:
        latency: 0.5
        time_format: "%%H:%%M:%%S"

teracy-dev:

  extensions:

    - _id: "entry-essential"
      path:
        extension: teracy-dev-essential
      location:
        git:
          remote:
            origin: https://github.com/teracyhq-incubator/teracy-dev-essential.git
          branch: develop
      require_version: ">= 0.3.0-SNAPSHOT"
      enabled: true


nodes:
  - _id: "0" # core-0
    vm:
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
  - _id: "entry-1"
    name: "%{node_name_prefix}-02"
    vm:
      hostname: "%{node_hostname_prefix}-02.%{node_domain_affix}"
      synced_folders:
        - _id: "entry-workspace"
          type: rsync
          host: "./workspace"
          guest: "/home/vagrant/workspace"
          rsync__exclude:
            - ".git"
            - ".idea"
            - ".#*"
          create: true
```

so expected: gatling-rsync-auto should run on both nodes.

For boolean value, it's best practices to use `TeracyDev::Util.true?`

**Comment by hieptranquoc** [comment]
@hoatle I've updated! Please check this PR again! Thank you!

File: lib/teracy-dev-essential/config/rsync_recovery.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
+require 'teracy-dev/config/configurator'
+require 'teracy-dev/plugin'
+
+require 'yaml'
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/rsync_recovery.rb]: remove this

File: lib/teracy-dev-essential/config/rsync_recovery.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+module TeracyDevEssential
+  module Config
+    class RsyncRecovery < TeracyDev::Config::Configurator
+
+      def configure_common(settings, config)
+        if gatling_rsync_installed?
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/rsync_recovery.rb]: move this after the vagrant version check

**Pull request merged**